app/src/main/java/com/android/streetworkapp/model/user/UserViewModel.kt

```diff
@@ -158,7 +158,14 @@ open class UserViewModel(private val repository: UserRepository) : ViewModel() {
    * @param points The number of points to add to the user's score.
    */
   fun increaseUserScore(uid: String, points: Int) =
-      viewModelScope.launch { repository.increaseUserScore(uid, points) }
+      viewModelScope.launch {
```

Comment:
It's dangerous that we need to update the user manually here. Would be better to return the user from repo.increaseUserScore and just store it inside currentUser instead

Reply:
I understand your concern. I think that the rework of the repo is outside the scope of this PR, we could try to make another version in the upcoming sprint.

Reply:
yeah but it should be resolved in another pr:
since we have
`        _currentUser.value?.let {
          val updatedUser = it.copy(score = it.score + points)
          _currentUser.value = updatedUser
        }`

in the viewmodel that gets executed even if firebase fails.
Someone could: turn off internet, spam an action that increases points, turn on internet again, go to progression and have checkscore being triggered. Then he will receive all the achievements even if he doesn't have the points.